### ISSUE TITLE
Document name qualification

### ISSUE THREAD

**Issue by jseldess** [cockroachdb/docs#1973]
A comment from @knz:

Usually a fully qualified name is written like this: `exampledb.customer.address` or `test.kv.v` and can contain at most one or two periods. For example:

```
create database ".𛲄·.·";
create table "𛲄.𛲄·"("𛲄.·.𛲄" int);
select "𛲄.𛲄·"."𛲄.·.𛲄" from ".𛲄·.·"."𛲄.𛲄·"."𛲄.·.𛲄";
```

**Comment by knz** [comment]
I'm not sure I was serious...

**Comment by jseldess** [comment]
@knz, what do you mean? I know those examples are a pretty ridiculous, but seems like not a bad idea to clarify something like:

> A qualified name uses a single period to indicate a `database.table` relationship or a `table.column` relationship. A fully qualified name can thus contain, at most, one or two periods, excluding periods used in the entity names themselves.

**Comment by knz** [comment]
All right. However I would not use the word "fully qualified" - this is verbiage from network tech, not database tech.

There are two things that can be named in SQL:

- relations (tables or views) in storage: these have a name of the form `[dbname.]name`. The `dbname.` part is optional, if omitted the "current database" is used. These relation names can be used in a FROM clause. For example `kv` and `t.kv` in `select * from kv, t.kv`
- column references: these have a name of the form `[relname.]name`. The `relname` part is optional, if omitted it is derived from context. These column references are used in value expressions. For example `v` and `kv.v` in `select v, kv.v from kv`. Note that `select t.kv.v from t.kv` is incorrect in PostgreSQL (postgres forgets the database prefix after recognizing the FROM clause so the column references cannot use it any more) but correct in CockroachDB as an extension.
- relations defined inside a query using WITH (not supported in CockroachDB yet). These do not have a database prefix. For example the name "foo" in `with foo as (select * from kv)  select * from foo`

**Comment by knz** [comment]
@benesch feel free to propose a phrasing for the explanation of zone naming that aligns with what SQL does otherwise, without using the adjective "qualified" :)

**Comment by benesch** [comment]
Heh, I'm not sure I share your objection to the term "qualified." `col` names a column, `tbl.col` is a qualified column name, but only `db.tbl.col` is fully qualified. Seems reasonable to me.

**Comment by jseldess** [comment]
I also don't understand the objection to `qualified`. It makes sense to me, but I trust that @knz is more sensitive to overlapping contexts and possible resulting confusions.

**Comment by knz** [comment]
I did a little research and apparently the word "Qualified" is used in the context of table names specifically; a name is "qualified" when it includes the database (in pg: "schema") prefix. A "qualified column reference" for example is one that includes the name of the database. There's no sign of "qualification" in the difference between "tablename.colname" and just "colname".

**Comment by knz** [comment]
This was fixed by #2748 / #2738.
